### PR TITLE
Feature/m3 game logic

### PR DESCRIPTION
**Pull request by didymus707**
Milestone 3: Game logic completed

### CODE REVIEW

**Review by imhta — CHANGES_REQUESTED** [review]
# Changes required 🛑 

### Hello  @didymus707 @jubaan  , 👋 You have done a great job in this project 👏 

* [x] The application should detect a sequence of three vertical elements that are equal

image




----
### Don't forgot to request for another code review
----
**Pon Muthu Selvam N**
__nponmuthuselvam@gmail.com__
_If you have any queries, feel free to ping me on slack_

**Comment by jubaan** [comment]
> # Changes required stop_sign
> ### Hello @didymus707 @jubaan , wave You have done a great job in this project clap
> 
>     * [x]  The application should detect a sequence of three vertical elements that are equal
> 
> 
> image
> ### Don't forgot to request for another code review
> 
> **Pon Muthu Selvam N**
> **[nponmuthuselvam@gmail.com](mailto:nponmuthuselvam@gmail.com)**
> _If you have any queries, feel free to ping me on slack_

Hi @imhta thanks for your review... I'm running the game but I can't get that error. This is what I get if get the marks in vertical line. 

image

**Comment by jubaan** [comment]
Could you be kindly enough to guide me, so I can replicate this error. It would be really appreciated :)

**Comment by didymus707** [comment]
I was about saying the same thing @jubaan . The game works well... See below
ttt

**Comment by didymus707** [comment]
Tried using some invalid options to check the game but it still works fine
ttt1
trying the invalid errors

**Comment by imhta** [comment]
@didymus707 @jubaan Most of the times it is working fine but if I am doing in a random way, it breaks don't know why? 
So, find and try to fix that.  below image represents all the input given to break the game.

If you are sure it won't break give another code review request

image

**Comment by jubaan** [comment]
> @didymus707 @jubaan Most of the times it is working fine but if I am doing in a random way, it breaks don't know why? find and try to fix that. below image represents all the input given to break the game.
> 
> image

Thanks a lot! We'll work on it, to see whats wrong!

**Comment by jubaan** [comment]
> @didymus707 @jubaan Most of the times it is working fine but if I am doing in a random way, it breaks don't know why? find and try to fix that. below image represents all the input given to break the game.
> 
> image

Thank you @imhta we could duplicate the error, and now the programs works at it suppose to.

**Comment by didymus707** [comment]
Winning match fixed

**Review by Forison — CHANGES_REQUESTED** [review]
### Status: Require changes🛑
## 🙋‍♂️ team
good job on addressing changes associated with the behavior of the application🙂 raised in the previous review.
Here are some other issues to address to get the project approved🙂
* [ ] No puts/print and get.chomp outside main.rb file ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

* [ ] It is better and cleaner to house each class in its own file and then use require_relative to import them when the need arises.
### That is it, kindly make these changes and submit the project for review
#### Happy coding💻
#### Addo Forison

File: bin/main.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require_relative '../lib/game_logic'
+
+puts "\n" + '*' * 50
```
Comment (Forison) [inline bin/main.rb]: 👉 this part of the code can be housed in another file and imported into main. Main should only serve as an entry point for your app execution🙂

**Comment by didymus707** [comment]
Milestone 3: All review changes made

**Review by Forison — CHANGES_REQUESTED** [review]
### Status: REquire changes🛑
## 🙋‍♂️ TEAM,
you have done a good job on this one so far,
Here are a few issues we have to address to get this project approved.🙂
* [x] The game has to end immediately there is a winner. currently, this what I get after winning👇
Screenshot from 2020-02-13 12-53-17

* [x] The game should inform players of a draw. Currently, this what I get when there is a draw.👇
Screenshot from 2020-02-13 12-55-06

### [optional]
You make split the classes in the logic file to three different files. This makes the code more readable.
## That is it kindly make these changes and submit the project for review.🙂
#### Happy coding
#### Addo Forison

**Review by fatymahmed — APPROVED** [review]
## Hi @jubaan @didymus707 
Your game works, great job on that 👏👏.
 - Remember you shouldn't have puts/print and get.chomp outside `main.rb` file, that means your logic shouldn't have them!. Please fix this before merging [screenshot from Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) 


### It's a 👍🏾 from me!
## ✔️ Approved!
Good luck with your next projects.
 _Don't forget to submit your project completion form_
_Feel free to reach out to me on slack(Fatima) if you have any questions, I'd be happy to help._

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    end
+  end
+
+  def space_selection(pick, mark, choices)
+    if @availible_spaces.include?(pick)
+      @availible_spaces[pick - 1] = mark
+      choices << pick
+    else
+      puts 'Please select a valid option'
+      print '> '
```
Comment (fatymahmed) [inline lib/board.rb:20]: - `print` and `puts` shouldn't be used here.

File: lib/board.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      choice = gets.chomp.to_i
+      choices << choice
```
Comment (fatymahmed) [inline lib/board.rb:22]: - `gets.chomp` shouldn't be used here

**Pull request closed without merging**